firmware/user/modes/mode_ddr.c

```diff
@@ -0,0 +1,876 @@
+            {
+                curRow->count--;
+                curRow->start = (curRow->start + 1) % ARROW_ROW_MAX_COUNT;
+                ddrHandleMiss();
+            }
+            
+        }
+
+        if (!ddr->isSongOver && canSpawnArrow && arrowsSpawnedThisBeat < 2)
+        {
+            if (rand() % 100 < percentChanceSpawn)
+            {
+                arrowsSpawnedThisBeat++;
+                curRow->arrows[(curRow->start+curRow->count) % ARROW_ROW_MAX_COUNT].hPos = 0;
+                curRow->count++;
+            }
+        }
+    }
+}
+
+/**
+ * @brief Called on a timer, this animates the note asset
+ *
+ * @param arg unused
+ */
+static void ICACHE_FLASH_ATTR ddrAnimateNotes(void* arg __attribute__((unused)))
+{
+    ddr->NoteIdx = (ddr->NoteIdx + 1) % 4;
+    // testUpdateDisplay();
+}
+
+static void ICACHE_FLASH_ATTR ddrUpdateDisplay(void* arg __attribute__((unused)))
+{
+    switch(ddr->mode)
+    {
+        default:
+        case DDR_MENU:
+        {
+            drawMenu(ddr->menu);
+            break;
+        }
+        case DDR_SCORE:
+        {
+            clearDisplay();
+            char perfectsText[24];
+            char hitsText[24];
+            char missesText[24];
+
+            ets_snprintf(perfectsText, sizeof(perfectsText),   "Perfects:    %03d", ddr->perfects);
+            ets_snprintf(hitsText, sizeof(perfectsText),       "Okays:       %03d", ddr->okays);
+            ets_snprintf(missesText, sizeof(perfectsText),     "Misses:      %03d", ddr->misses);
+
+            if (ddr->didLose)
+            {
+                plotText(38,5, "You died", IBM_VGA_8, WHITE);
+            }
+            else
+            {
+                plotText(38,5, "You win!", IBM_VGA_8, WHITE);
+            }
+
+            plotText(36,30, perfectsText, TOM_THUMB, WHITE);
+            plotText(36,40, hitsText, TOM_THUMB, WHITE);
+            plotText(36,50, missesText, TOM_THUMB, WHITE);
+            break;
+        }
+        case DDR_GAME:
+        {
+            //Game update logic
+            ddrHandleArrows();
+
+            // Clear the display
+            clearDisplay();
+
+            //ddrUpdateButtons();
+            ddrArrowRow* curRow;
+            ddrArrow* curArrow;
+            int curStart;
+            int curCount;
+            int curEnd;
+
+            for (int rowIdx = 0; rowIdx < 4; rowIdx++)
+            {
+                curRow = &(ddr->arrowRows[rowIdx]);
+                curStart = curRow->start;
+                curCount = curRow->count;
+                curEnd = (curStart + curCount) % ARROW_ROW_MAX_COUNT;
+
+                for(int arrowIdx = curStart; arrowIdx != curEnd; arrowIdx = (arrowIdx + 1) % ARROW_ROW_MAX_COUNT)
+                {
+                    curArrow = &(curRow->arrows[arrowIdx]);
+                    
+                    
+                    drawPngSequence(&ddrSkullSequenceHandle, (curArrow->hPos-400)/12, 48 - rowIdx * 16, false, false, 0, ddr->NoteIdx);
```

Comment:
I had a tough time playing the first few times and I realized it's because all the skulls point the same way. The sprites are all 15x15 (though the actual image is only 14x14). They're nice and square, which should make rotating them like DDR arrows a cinch. There's even an easy rotate argument for you to use!